File: multilanguage/graalvm_poc/src/main/ruby/student.rb
#!/usr/bin/ruby

class Student
    def initialize
        @name = ""
        @id = 0
    end

    def setName(name)
        @name = name
    end

    def setId(id)
        @id = id
    end

    def getName
        return @name
    end

    def getID
       return @id
    end
end

File: sapphire/sapphire-core/src/test/resources/student.js
class Student {
    constructor() {
        this.id = 0;
        this.name = "";
        this.buddies = [];
    }

    setId(id) {
        this.id = id;
    }

    getId() {
        return this.id;
    }

    setName(name) {
        this.name = name;
    }

    getName() {
        return this.name;
    }

    addBuddy(buddy) {
        this.buddies.push(buddy);
    }

    getBuddies() {
        return this.buddies;
    }
}
